### PR TITLE
Add missing Exchange properties and method definitions

### PR DESCRIPTION
**Pull request by omidkrad**
I was not sure about some function signatures, so I just added their stubs with no function signature. Plus added `Dictionary` type, sorted methods alphabetically and some more fixes.

### CODE REVIEW

File: ccxt.d.ts [hunk 1]
```diff
@@ -272,12 +332,53 @@ declare module 'ccxt' {
...
         substituteCommonCurrencyCodes: boolean;
         timeframes: any;
-        has: { [what: string]: any }; // https://github.com/ccxt/ccxt/pull/1984
+        has: Dictionary<any>; // https://github.com/ccxt/ccxt/pull/1984
         balance: object;
         orderbooks: object;
         orders: object;
         trades: object;
         userAgent: { 'User-Agent': string } | false;
+        limits: { amount: MinMax, price: MinMax, cost: MinMax };
```
Comment (omidkrad) [inline ccxt.d.ts]: I'm not sure why Exchange has this property. For binance it returns:
```js
{
  amount: { min: undefined, max: undefined },
  price: { min: undefined, max: undefined }, 
  cost: { min: undefined, max: undefined }   
}
```

Comment (kroitor) [reply ccxt.d.ts]: @omidkrad it's the exchange-wide defaults that get extended and overrided for each particular market. Some exchanges don't have min limits defined anywhere in their API, therefore we have to define their limits for the entire exchange, and that structure gets inherited by every market. Binance does have min limit info in the market APIs themselves, therefore the values of this exchange-wide property is undefined – it gets defined upon `fetchMarkets()/loadMarkets()`.

**Comment by omidkrad** [comment]
Fixed merge conflicts.

**Comment by omidkrad** [comment]
Fixed merge conflicts.

File: ccxt.d.ts [hunk 2]
```diff
@@ -220,6 +220,54 @@ declare module 'ccxt' {
...
+            adjustForTimeDifference: boolean;
+            parseOrderToPrecision: boolean;
+            newOrderRespType: {
+                market: 'FULL' | string;
+                limit: 'RESULT' | string;
+            };
+        };
+        urls: {
+            logo: string;
+            api: {
```
Comment (blair) [inline ccxt.d.ts]: Is this correct? Looking at coinbasepro.js it's a single URL. Also, https://github.com/ccxt/ccxt/wiki/Manual says:

> urls['api']: The single string literal base URL for API calls or an associative array of separate URLs for private and public APIs.

File: ccxt.d.ts [hunk 3]
```diff
@@ -212,9 +215,12 @@ declare module 'ccxt' {
         tag?: string;
     }
 
-    // timestamp, open, high, low, close, volume
+    /** [ timestamp, open, high, low, close, volume ] */
     export type OHLCV = [number, number, number, number, number, number];
 
+    /** Request parameters */
+    type params = Dictionary<string | number>;
```
Comment (blair) [inline ccxt.d.ts]: Suggest making this uppercase so it's clear it's a type. It's odd reading it below in the function args as lowercase.

File: ccxt.d.ts [hunk 4]
```diff
@@ -381,77 +381,75 @@ declare module 'ccxt' {
...
+        fetchOpenOrders (symbol?: string, since?: number, limit?: number, params?: params): Promise<Order[]>;
         fetchOrder (id: string, symbol: string, params?: params): Promise<Order>;
         fetchOrderBook (symbol: string, limit?: number, params?: params): Promise<OrderBook>;
+        fetchOrderBooks (...args: any): Promise<any>; // TODO: add function signatures
+        fetchOrders (symbol?: string, since?: number, limit?: number, params?: params): Promise<Order[]>;
+        fetchOrderStatus (id: string, market: string): Promise<string>;
+        fetchStatus (...args: any): Promise<any>; // TODO: add function signatures
         fetchTicker (symbol: string, params?: params): Promise<Ticker>;
         fetchTickers (symbols?: string[], params?: params): Promise<Dictionary<Ticker>>;
+        fetchTime (...args: any): Promise<any>; // TODO: add function signatures
```
Comment (blair) [inline ccxt.d.ts]: Since you're here, what about changing this `Promise<any>` to `Promise<number>` ?

File: ccxt.d.ts (hunk 2 above)
Comment (omidkrad) [reply ccxt.d.ts]: Fixed.

File: ccxt.d.ts (hunk 3 above)
Comment (omidkrad) [reply ccxt.d.ts]: Done.

File: ccxt.d.ts (hunk 4 above)
Comment (omidkrad) [reply ccxt.d.ts]: Done.

**Comment by blair** [comment]
Thanks for so quickly making the changes! Hopefully this will be merged soon by the author!

**Comment by omidkrad** [comment]
Fixed merge conflicts again. I think to review it's easier to review commits individually because the last commit "Sort methods alphabetically" makes the diff hard to review.

**Comment by omidkrad** [comment]
Fixed merge conflicts again. I think to review it's easier to review commits individually because the last commit "Sort methods alphabetically" makes the diff hard to review.

**Comment by omidkrad** [comment]
Awesome! Thanks for merging.

**Pull request merged**